### PR TITLE
Validation of HPA custom metrics annotation

### PR DESCRIPTION
**Pull request by mwielgus**

### CODE REVIEW

**Comment by mwielgus** [comment]
cc: @fgrzadkowski @piosz

File: pkg/apis/extensions/validation/validation.go [hunk 1]
```diff
@@ -83,9 +85,34 @@ func ValidateSubresourceReference(ref extensions.SubresourceReference, fldPath *
...
+			}
+		}
+	}
+	return allErrs
+}
+
 func ValidateHorizontalPodAutoscaler(autoscaler *extensions.HorizontalPodAutoscaler) field.ErrorList {
 	allErrs := apivalidation.ValidateObjectMeta(&autoscaler.ObjectMeta, true, ValidateHorizontalPodAutoscalerName, field.NewPath("metadata"))
 	allErrs = append(allErrs, validateHorizontalPodAutoscalerSpec(autoscaler.Spec, field.NewPath("spec"))...)
+	allErrs = append(allErrs, validateHorizontalPodAutoscalerAnnotations(autoscaler.Annotations, field.NewPath("metadata"))...)
```
Comment (wojtek-t) [inline pkg/apis/extensions/validation/validation.go]: should "metadata" be "annotations" ?

**Comment by wojtek-t** [comment]
LGTM - just one minor nit

File: pkg/apis/extensions/validation/validation.go (hunk 1 above)
Comment (mwielgus) [inline pkg/apis/extensions/validation/validation.go]: annotations are in metadata

**Pull request merged**